catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -155,7 +164,9 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
                 List<Metacard> createdMetacards = createResponse.getCreatedMetacards();
 
                 for (Metacard metacard : createdMetacards) {
-                    fileIdMap.store(key, metacard.getId());
+                    if (safeKey != null) {
```

Comment:
yeah @stustison that sounds better to me. Seems like the code should be intentional on whether it is logging for debugging purposes or looping through the created metacards for another reason.